### PR TITLE
[0442] Fixed rubocop issues part 1

### PR DESCRIPTION
**Pull request by defong**
### Context
Our code base is really filthy

### Changes proposed in this pull request
Fixed the low hanging fruits

### Guidance to review
Best to review per commit, with the first one being auto corrected and the following commits fixing per rule.
Can ignore `codeclimate`

#### Before
```bash
823 files inspected, 3316 offenses detected, 879 offenses auto-correctable
```
#### After
```bash
823 files inspected, 2457 offenses detected
```
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[Clear rubocop_todo part 1](https://trello.com/c/scLVGEHJ/442-clear-rubocoptodo-part-1)

**Review by george-bit — APPROVED** [review]
some sensible changes for consistency, might be an idea for others to take a look too just to get familiar with the new rules 🤖 🧑‍⚖️ 

...on the other hand that was a long review 🥵,  so go ahead an merge if you want!

File: app/services/provider_reporting_service.rb [hunk 1]
```diff
@@ -42,8 +42,16 @@ def group_by_count(column)
     closed = @closed_providers.group(column).count
 
     {
-      open: Provider.send(column.to_s.pluralize).map { |key, _value| x = {}; x[key.to_sym] = open[key] || 0; x }.reduce({}, :merge),
-      closed: Provider.send(column.to_s.pluralize).map { |key, _value| x = {}; x[key.to_sym] = closed[key] || 0; x }.reduce({}, :merge),
+      open: Provider.send(column.to_s.pluralize).map { |key, _value|
```
Comment (darokel) [inline app/services/provider_reporting_service.rb:45]: I'm surprised Rubocop didn't complain about using `do end` for multi-line blocks. Do we need to add a rule to enforce this?

Comment (defong) [reply app/services/provider_reporting_service.rb:45]: Sure, that can that be for another time...

**Review by malcolmbaig — APPROVED** [review]

**Pull request merged**